src/uucore/src/lib/features/sum.rs

```diff
@@ -0,0 +1,486 @@
+// spell-checker:ignore memmem algo
+//! Implementations of digest functions, like md5 and sha1.
+//!
+//! The [`Digest`] trait represents the interface for providing inputs
+//! to these digest functions and accessing the resulting hash. The
+//! [`DigestWriter`] struct provides a wrapper around [`Digest`] that
+//! implements the [`Write`] trait, for use in situations where calling
+//! [`write`] would be useful.
+use std::io::Write;
+
+use hex::encode;
+#[cfg(windows)]
+use memchr::memmem;
+
+pub trait Digest {
+    fn new() -> Self
+    where
+        Self: Sized;
```

Comment:
I think I can't do this, because a `Sized` trait can't be used as object.
See this article
https://stackoverflow.com/questions/70377145/this-trait-cannot-be-made-into-an-object-because-it-requires-self-sized-ex

Reply:
Oh yeah I forgot. The `new` and `reset` methods are annoying here. I guess it's fine then.